tests/helpers/ztest.py

```diff
@@ -47,6 +48,96 @@ def from_yaml_file(cls, path):
+                interpreter_str = helper.build_interpreter_string()
+                inventory = helper.get_inventory_info()
+                python_path = helper.get_python_path()
+        Shell example with pytest:
+            pytest tests/functional/modules/test_zos_mount_func.py::test_basic_mount --host-pattern=all -s -v --zinventory-raw='{"host": "zvm.ibm.com", "user": "ibmuser", "zoau": "/zoau/v1.3.1", "pyz": "/allpython/3.10/usr/lpp/IBM/cyp/v3r10/pyz", "pythonpath": "/zoau/v1.3.1/lib/3.10", "extra_args":{"volumes":["222222","000000"],"other":"something else"}}' -s
+
+        {
+            "host":"zvm.ibm.com",
+            "user":"ibmuser",
+            "zoau":"/zoau/v1.3.1",
+            "pyz":"/allpython/3.10/usr/lpp/IBM/cyp/v3r10/pyz",
+            "pythonpath": "/zoau/v1.3.1/lib/3.10",
+            "extra_args":{
+                "volumes":[
+                    "vol1",
+                    "vol2"
+                ],
+                "other": "something else" }
+        }
+        """
+        #TODO: add support for a positional string, eg "host,user,zoau,pyz" then convert it as needed
+
+        host, user, zoau, pyz, pythonpath, extra_args = None, None, None, None, None, None
+
+        src = json.loads(src)
+        # Traverse the src here , can we trow an exception?
+        for key, value in src.items():
+            if key == "host":
+                host = value
+            elif key == "user":
+                user = value
+            elif key == "zoau":
+                zoau = value
+            elif key == "pyz":
+                pyz = value
+            elif key == "pythonpath":
+                pythonpath = value
+            elif key == "extra_args":
+                extra = value
+
+        for prop in [host, user, zoau, pyz, pythonpath]:
+            if prop is None:
+                message = f"Invalid value for use with keyword, the value must not be None"
+                raise ValueError(message)
+
+        environment_vars = dict()
+        environment_vars.update({'_BPXK_AUTOCVT': 'ON'})
+        environment_vars.update({'_CEE_RUNOPTS': '\'FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)\''})
+        environment_vars.update({'_TAG_REDIR_IN': 'txt'})
+        environment_vars.update({'_TAG_REDIR_OUT': 'txt'})
+        environment_vars.update({'LANG': 'C'})
+        environment_vars.update({'ZOAU_HOME': zoau})
+        environment_vars.update({'LIBPATH': f"{zoau}/lib:{pyz}/lib:/lib:/usr/lib:."})
+        environment_vars.update({'PYTHONPATH': f"{pythonpath}"}) # type: ignore
+        environment_vars.update({'PATH': f"{zoau}/bin:{pyz}/bin:/bin:/usr/sbin:/var/bin"})
+        environment_vars.update({'PYTHONSTDINENCODING': 'cp1047'})
+
+        testvars = dict()
+        testvars.update({'host': host})
+        testvars.update({'user': user})
+        testvars.update({'python_path': f"{pyz}/bin/python3"})
+        testvars.update({'environment': environment_vars})
+
+        if(extra):
```

Comment:
You might get a NameErrror here if key != `extra_args` bc `extra` is not initialized.

Reply:
Fixed, thanks.